File: Examples/test-suite/ocaml/char_binary_rev_len_runme.ml
open Swig
open Char_binary_rev_len

let _ =
  let t = new_Test '() in
  assert (t -> strlen ("hile") as int = 4);
  assert (t -> strlen ("hil\x00") as int = 4);
  let t2 = new_Test '("hile") in
  assert (t2 -> getisize () as int = 4);
  assert (t2 -> strlen ("hil\x00a") as int = 5);
;;

File: Examples/test-suite/go/char_binary_rev_len_runme.go
package main

import . "swigtests/char_binary_rev_len"

func main() {
	t := NewTest()
	defer DeleteTest(t)

	if t.Strlen("hile") != 4 {
		panic("bad multi-arg typemap 1")
	}

	if t.Strlen("hil\000") != 4 {
		panic("bad multi-arg typemap 2")
	}

	t2 := NewTest("hile")
	defer DeleteTest(t2)

    if t2.Getisize() != 4 {
		panic("bad multi-arg typemap 3")
    }

	if t2.Strlen("hil\000a") != 5 {
		panic("bad multi-arg typemap 4")
	}
}
